libr/anal/fcn.c

```diff
@@ -1009,9 +1087,12 @@ static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut6
 				}
 			}
 			if (anal->opt.jmptbl) {
-				ut64 jmp_addr = 0;
-				if (is_delta_pointer_table (anal, op.addr, op.ptr, &jmp_addr)) {
-					ret = try_walkthrough_jmptbl (anal, fcn, depth, jmp_addr, op.ptr, 4, 4);
+				RAnalOp jmp_aop = {0};
+				if (is_delta_pointer_table (anal, fcn, op.addr, op.ptr, &jmp_aop)) {
```

Comment:
could we have jmp table shorter than 4 entries? I think `is_delta_pointer_table` tries to validate 4 entries in the jmp table. Would it make sense to try to first get the jmptbl_info and then checking if the first `n` entries are valid?

Reply:
1) Jump table size less than 4. I haven't been able to get a compiler to generate one with that few entries but I guess one could reduce the sanity check to 2 entries.
2) Sanity test to exit early makes sense from a performance pov. E.g. looking at [this](https://drive.google.com/open?id=1BkMo4ZXNS3K5_yl0yrHlqY1BiOz4EwHV), where determining the size is just overhead since the lea ptr is not valid

Reply:
I see, thanks for the reply!